TLM/TLM/Custom/PathFinding/CustomPathManager.cs

```diff
@@ -97,7 +133,46 @@ public class CustomPathManager : PathManager {
                 }
             }
 
-            InitDone = true;
+            Log._Debug("UpdateWithPathManagerValues success");
+        }
+
+        public void UpdateOldPathManagerValues(PathManager stockPathManager) {
+            stockPathManager.m_simulationProfiler = m_simulationProfiler;
+            stockPathManager.m_drawCallData = m_drawCallData;
+            stockPathManager.m_properties = m_properties;
+            stockPathManager.m_pathUnitCount = m_pathUnitCount;
+            stockPathManager.m_renderPathGizmo = m_renderPathGizmo;
+            stockPathManager.m_pathUnits = m_pathUnits;
+            stockPathManager.m_bufferLock = m_bufferLock;
+
+            int n = _replacementPathFinds.Length;
+
+            Log._Debug("Creating " + n + " stock PathFind objects.");
+            PathFind[] stockPathFinds = new PathFind[n];
+
+            FieldInfo f_pathfinds = typeof(PathManager).GetField(
+                  "m_pathfinds",
+                  BindingFlags.NonPublic | BindingFlags.Instance)
+                  ?? throw new Exception("f_pathFinds is null");
+
+            lock(m_bufferLock) {
+                for(int i = 0; i < n; i++) {
+                    stockPathFinds[i] = gameObject.AddComponent<PathFind>();
+                }
+
+                Log._Debug("Setting stockPathFinds to stock collection");
+                f_pathfinds?.SetValue(stockPathManager, stockPathFinds);
+
+                for(int i = 0; i < n; i++) {
+                    Log._Debug($"PF {i}: {_replacementPathFinds[i].m_queuedPathFindCount} queued path-finds");
+
+                    // would cause deadlock since we have a lock on m_bufferLock
+                    // customPathFinds[i].WaitForAllPaths();
+                    Destroy(_replacementPathFinds[i]);
+                }
+            }
+
+            Log._Debug("UpdateOldPathManagerValues success");
```

Comment:
is this message useful? Can it fail?

Reply:
it failed a few times for me until I fixed it.